openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -244,27 +268,62 @@ def _get_profile_image(self, user_profile):
     def update(self, instance, validated_data):
         """
         Update the profile, including nested fields.
+
+        Raises:
+        errors.AccountValidationError: the update was not attempted because validation errors were found with
+            the supplied update
         """
         language_proficiencies = validated_data.pop("language_proficiencies", None)
 
         # Update all fields on the user profile that are writeable,
-        # except for "language_proficiencies", which we'll update separately
-        update_fields = set(self.get_writeable_fields()) - set(["language_proficiencies"])
+        # except for "language_proficiencies" and "social_links", which we'll update separately
+        update_fields = set(self.get_writeable_fields()) - set(["language_proficiencies"]) - set(["social_links"])
         for field_name in update_fields:
             default = getattr(instance, field_name)
             field_value = validated_data.get(field_name, default)
             setattr(instance, field_name, field_value)
 
-        instance.save()
-
-        # Now update the related language proficiency
+        # Update the related language proficiency
         if language_proficiencies is not None:
             instance.language_proficiencies.all().delete()
             instance.language_proficiencies.bulk_create([
                 LanguageProficiency(user_profile=instance, code=language["code"])
                 for language in language_proficiencies
             ])
 
+        # Update the user's social links
+        social_link_data = self._kwargs['data']['social_links'] if 'social_links' in self._kwargs['data'] else None
+        if social_link_data is not None and len(social_link_data) > 0:
```

Comment:
@HarryRein: Is this equivalent to `if social_link_data:` in Python?

Reply:
Not the exact same, but in this case if social_link_data works. I was explicitly calling is not None because it was assigned None in the line before, I'll change it.